gatsby-node.js

```diff
@@ -1,95 +1,129 @@
-const axios = require('axios');
-const matter = require('gray-matter');
-const fs = require('fs-extra');
-const path = require('path');
-
-const { createRemoteFileNode  } = require(`gatsby-source-filesystem`)
-
-exports.onPreInit = () => console.log("Loaded gatsby-source-onlinesales plugin")
-
-const POST_NODE_TYPE = `Post`
-
-exports.sourceNodes = async ({
-  actions,
-  createContentDigest,
-  createNodeId,
-  getCache,
-  cache,
-
-}, options) => {
-  const { createNode } = actions
-  const processRemoteImage = async (url, folder) => {
-    const nodeId = createNodeId(`RemoteImage-${url}`);
-    const fileNode = await createRemoteFileNode({
-        url,
-        parentNodeId: nodeId,
-        getCache,
-        createNode,
-        createNodeId,
-        cache,
-    });
-    const fileName = `${fileNode.name}${fileNode.ext}`;
-    const outputPath = path.join(process.cwd(), 'public', 'static', folder);
-    const filePath = path.join(outputPath, fileName);
-    await fs.ensureDir(outputPath);
-    await fs.copy(fileNode.absolutePath, filePath);
-    const readyPath = `/static/${folder}/${fileName}`;
-    return readyPath;
-  };
+const axios = require("axios");
+const matter = require("gray-matter");
+
+const extractImages = require("./extractImages");
+const processRemoteImage = require("./processRemoteImage");
+const {
+  PAGE_NODE_TYPE,
+  POST_NODE_TYPE,
+  IMAGE_FILE_TYPE,
+} = require("./constants");
+
+exports.onPreInit = () =>
+  console.log("Loaded gatsby-source-onlinesales plugin");
+
+const getNodeType = (type) =>
+  type === "post" ? POST_NODE_TYPE : PAGE_NODE_TYPE;
+
+const getNodeLanguage = (language) =>
```

Comment:
@Yuri2b it is OK to keep it this way for now, but moving forward we should consider there could be other languages than ru and en, what happens than?

Reply:
I've added simple parsing for 2 chars language code, assuming that CMS provides ISO 639-1 or BCP 47 languge code format (e.g. "en" or "en-US")